### PR TITLE
Clippy changes

### PR DESCRIPTION
**Pull request by MrRobotMan**
I made quite a few changes to the codebase, especially the XXX_generator.rs files in order to get everything to pass clippy.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
```
Comment (brettfo) [inline .vscode/settings.json:1]: Why were these files deleted?  Do they interfere with clippy somehow?  If possible I'd like to keep them around.

Comment (MrRobotMan) [reply .vscode/settings.json:1]: They were deleted because I failed to realize they were part of the original repo. I had added my own .vscode settings and then removed them without realizing I had removed the ones from you.

**Comment by waywardmonkeys** [comment]
@MrRobotMan @brettfo It would be great to see some or all of this land ...

@MrRobotMan Do you plan to update the PR to restore the `.vscode` files and resolve the conflicts?

**Comment by MrRobotMan** [comment]
@waywardmonkeys I just restored the original .vscode/settings.json file and merged the conflicts.

File: .vscode/settings.json (hunk 1 above)
Comment (MrRobotMan) [reply .vscode/settings.json]: Fixed with commit [0efd2b4](https://github.com/ixmilia/dxf-rs/pull/42/commits/0efd2b4a70c995f639be694f440151ea7b65a267)

**Comment by waywardmonkeys** [comment]
@MrRobotMan Cool ... up to @brettfo now then!
